lib/src/simple_tokenizer.dart

```diff
@@ -0,0 +1,200 @@
+    _scanner.scan(_commentEnd);
+    _state = _NgSimpleScannerState.text;
+    return new NgSimpleToken.commentEnd(offset);
+  }
+
+  NgSimpleToken scanElement() {
+    int offset = _scanner.position;
+    if (_scanner.peekChar() == null) {
+      return new NgSimpleToken.EOF(offset);
+    }
+    if (_scanner.scan(_allElementMatches)) {
+      Match match = _scanner.lastMatch;
+      if (matchesGroup(match, 1)) {
+        return new NgSimpleToken.closeBracket(offset);
+      }
+      if (matchesGroup(match, 2)) {
+        return new NgSimpleToken.bang(offset);
+      }
+      if (matchesGroup(match, 3)) {
+        return new NgSimpleToken.dash(offset);
+      }
+      if (matchesGroup(match, 4)) {
+        return new NgSimpleToken.closeParen(offset);
+      }
+      if (matchesGroup(match, 5)) {
+        _state = _NgSimpleScannerState.text;
+        return new NgSimpleToken.tagEnd(offset);
+      }
+      if (matchesGroup(match, 6)) {
+        return new NgSimpleToken.forwardSlash(offset);
+      }
+      if (matchesGroup(match, 7)) {
+        return new NgSimpleToken.openBracket(offset);
+      }
+      if (matchesGroup(match, 8)) {
+        return new NgSimpleToken.openParen(offset);
+      }
+      if (matchesGroup(match, 9)) {
+        return new NgSimpleToken.whitespace(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 10)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 11)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 12)) {
+        return new NgSimpleToken.doubleQuotedText(
+            offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 14)) {
+        return new NgSimpleToken.singleQuotedText(
+            offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 16)) {
+        return new NgSimpleToken.doubleQuote(offset);
+      }
+      if (matchesGroup(match, 17)) {
+        return new NgSimpleToken.singleQuote(offset);
+      }
+      if (matchesGroup(match, 18)) {
+        return new NgSimpleToken.tagStart(offset);
+      }
+      if (matchesGroup(match, 19)) {
+        return new NgSimpleToken.equalSign(offset);
+      }
+      if (matchesGroup(match, 20)) {
+        return new NgSimpleToken.star(offset);
+      }
+    }
+    return new NgSimpleToken.unexpectedChar(
+        offset, new String.fromCharCode(_scanner.readChar()));
+  }
+
+  NgSimpleToken scanText() {
+    int offset = _scanner.position;
+    if (_scanner.peekChar() == null || _scanner.rest.length == 0) {
+      return new NgSimpleToken.EOF(offset);
+    }
+    if (_scanner.scan(_allTextMatches)) {
+      Match match = _scanner.lastMatch;
+      if (matchesGroup(match, 1)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 2)) {
+        _state = _NgSimpleScannerState.comment;
```

Comment:
Probably worth scanning the whole comment here rather than splitting it up into "open comment, comment text, close comment." Will make it easier in the body of the parser to step past comments if it can always just say "next()" rather than having to go past up to three times and check for EOF along the way.

Errors here (such as unclosed comments) should be reported here rather than in the parser for the same reason